Mage.Sets/src/mage/cards/f/FleshDuplicate.java

```diff
@@ -0,0 +1,58 @@
+package mage.cards.f;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.effects.common.CopyPermanentEffect;
+import mage.abilities.keyword.VanishingAbility;
+import mage.constants.SubType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.util.functions.CopyApplier;
+
+/**
+ *
+ * @author Skiwkr
+ */
+public final class FleshDuplicate extends CardImpl {
+
+    public FleshDuplicate(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{U}{U}");
+        
+        this.subtype.add(SubType.SHAPESHIFTER);
+        this.subtype.add(SubType.REBEL);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(0);
+
+        // You may have Flesh Duplicate enter the battlefield as a copy of any creature on the battlefield, except it has vanishing 3 if that creature doesn't have vanishing.
+        this.addAbility(new EntersBattlefieldAbility(
+                new CopyPermanentEffect(StaticFilters.FILTER_PERMANENT_CREATURE, new FleshDuplicateCopyApplier()),
+                true
+        ));
+    }
+
+    private FleshDuplicate(final FleshDuplicate card) {
+        super(card);
+    }
+
+    @Override
+    public FleshDuplicate copy() {
+        return new FleshDuplicate(this);
+    }
+}
+
+class FleshDuplicateCopyApplier extends CopyApplier {
+
+    @Override
+    public boolean apply(Game game, MageObject blueprint, Ability source, UUID copyToObjectId) {
+        if (blueprint.hasAbility(new VanishingAbility(3),game)) {
```

Comment:
this seems wrong.
- Shouldn't you be adding if it _doesn't_ have a vanishing ability yet?
- Need to check for any instance of vanishing, not specifically vanishing 3

Reply:
Yeah I meant to add some context to that in my PR and totally forgot to do that.  This was intended to be a workaround for the fact that VanishingAbility doesn't have a getInstance method to pass into hasAbility, but then when I messed up the check, it looked like it worked.  

What would be the better way to implement this functionality, modify VanishingAbility to have a blank instance call to pass back with getInstance, or to figure out a way to check for the sub ability VanishingTriggeredAbility and look for that instead?